tests/MongoDB.Driver.Core.Tests/Specifications/connection-monitoring-and-pooling/ConnectionMonitoringAndPoolingTestRunner.cs

```diff
@@ -0,0 +1,607 @@
+                var connectionPoolSettings = actualEvent.ConnectionPoolSettings();
+                if (expectedOption.IsInt32 && expectedOption == 42) // 42 - placeholder
+                {
+                    connectionPoolSettings.Should().NotBeNull();
+                }
+                else
+                {
+                    var expectedMaxPoolSize = expectedOption["maxPoolSize"].ToInt32();
+                    connectionPoolSettings.MaxConnections.Should().Be(expectedMaxPoolSize);
+                    var expectedMinPoolSize = expectedOption["minPoolSize"].ToInt32();
+                    connectionPoolSettings.MinConnections.Should().Be(expectedMinPoolSize);
+                    // maxIdleTimeMS : todo:
+                }
+            }
+
+            if (expectedEvent.TryGetValue("address", out var address))
+            {
+                var expectedAddress = address.ToInt32();
+                if (expectedAddress == 42)
+                {
+                    actualEvent.ServerId().EndPoint.Should().NotBeNull();
+                }
+                else
+                {
+                    // not expected code path
+                    throw new NotImplementedException();
+                }
+            }
+
+            if (expectedEvent.TryGetValue("reason", out var reason))
+            {
+                if (!(actualEvent is ConnectionClosedEvent)) // we don't support a reason for ConnectionClosedEvent
+                {
+                    actualEvent.CloseReason().Should().NotBeNull();
+                    actualEvent.CloseReason().ToString().ToLower().Should().Be(reason.ToString().ToLower());
+                }
+            }
+        }
+
+        private void AssertEvents(BsonDocument test, EventCapturer eventCapturer)
+        {
+            var actualEvents = GetFilteredEvents(eventCapturer, test);
+            var expectedEvents = GetExpectedEvents(test);
+            var minCount = Math.Min(actualEvents.Count, expectedEvents.Count);
+            for (var i = 0; i < minCount; i++)
+            {
+                var expectedEvent = expectedEvents[i];
+                JsonDrivenHelper.EnsureAllFieldsAreValid(expectedEvent, "type", "address", "connectionId", "options", "reason");
+                AssertEvent(actualEvents[i], expectedEvent);
+            }
+
+            if (actualEvents.Count < expectedEvents.Count)
+            {
+                throw new Exception($"Missing event: {expectedEvents[actualEvents.Count]}.");
+            }
+
+            if (actualEvents.Count > expectedEvents.Count)
+            {
+                throw new Exception($"Unexpected event of type: {actualEvents[expectedEvents.Count].GetType().Name}.");
+            }
+        }
+
+        private Task CreateTask(Action action)
+        {
+            return Task.Factory.StartNew(
+                action,
+                CancellationToken.None,
+                TaskCreationOptions.None,
+                new ThreadPerTaskScheduler());
```

Comment:
I really don't understand why `ThreadPerTaskSchedule` is needed, but I also don't see that it causes any harm.

Reply:
I changed this to:

    return Task.Factory.StartNew(
        action,
        CancellationToken.None);

and all tests still pass.

I don't think there is any harm in using `ThreadPerTaskScheduler` but I still don't understand why you think it is necessary.

Reply:
Discussed offline. Leaving it as is, but additionally, I've added a comment with an explanation of the idea of this task